main.py

```diff
@@ -179,24 +179,18 @@ def chooser_callback(self,uri_list):
         '''
         try:
             ss = SharedStorage()
-            for uri in uri_list: # only one list
-                
-                # copy to private cache
-                path = ss.copy_from_shared(uri)
-
-                #read content from cache
-                try:
-                    with open(path,'r',encoding = 'utf-8') as fread:
-                        reading = fread.read()
-
-                    #Clock is needed otherwise throw thread error
-                    Clock.schedule_once(lambda dt: self.populate_label(reading))
+            # copy to private cache
+            path = ss.copy_from_shared(uri[0])
+            
+            #read content from cache
+            with open(path,'r',encoding = 'utf-8') as fread:
+                reading = fread.read()
 
-                except Exception as e:
-                    toast(f"1.cannot open file path : {str(path)}  because : {str(e)}")
+            #Clock is needed otherwise throw thread error
+            Clock.schedule_once(lambda dt: self.populate_label(reading))
 
-        except Exception as e:
-            toast("something wrong with package androidstorage4kivy" + str(e))
+        except (FileNotFoundError, UnicodeError) as e:
+            toast(f"1.cannot open file path : {str(path)}  because : {str(e)}")
 
     def populate_label(self,reading):
```

Comment:
@javixav import `mainthread` from kivy.clock
and add the `@mainthread` decorator at the top of the function instead of using `Clock` to call this function